### PR TITLE
Appveyor

### PR DESCRIPTION
**Pull request by chcg**
- added appveyor.yml config
- see appveyor builds at https://ci.appveyor.com/project/chcg/nppgtags/build/4.2.1.7
- mingw x64 build is just a x86 build, because appveyor has no installation of the x86_64 version but it is easier to build it again than to remove it from the build matrix
- adapted cmake for win32 builds
- fixed compiler warnings
- updated n++ headers and merged changes for struct FuncItem

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -56,20 +69,25 @@ set (project_sources
...
 )
 
-find_library (comctl32
-    NAMES libcomctl32.a
-    PATHS ${win32_lib_dir}
-)
-
 add_definitions (${defs})
 
 include_directories (${win32_inc_dir})
```
Comment (pnedev) [inline CMakeLists.txt]: That should be moved in the `if (MINGW)` section below, correct?
Thanks.

**Comment by pnedev** [comment]
Thanks Christian, looks OK except for the comment above.
I have fixed some compile warnings I saw in your AppVeyor build.

File: CMakeLists.txt (hunk 1 above)
Comment (chcg) [inline CMakeLists.txt]: Yes, that makes just sense there. I will update.

**Comment by pnedev** [comment]
Great, thank you.
BR

**Pull request merged**